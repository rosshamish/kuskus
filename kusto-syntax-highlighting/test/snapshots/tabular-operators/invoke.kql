// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/invokeoperator

// clipped_average(): calculates percentiles limits, and then makes another 
//                    pass over the data to calculate average with values inside the percentiles
let clipped_average = (T:(x: long), lowPercentile:double, upPercentile:double)
{
   let high = toscalar(T | summarize percentiles(x, upPercentile));
   let low = toscalar(T | summarize percentiles(x, lowPercentile));
   T 
   | where x > low and x < high
   | summarize avg(x) 
};
range x from 1 to 100 step 1
| invoke clipped_average(5, 99)
