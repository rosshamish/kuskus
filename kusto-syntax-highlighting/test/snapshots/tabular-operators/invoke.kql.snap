>// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/invokeoperator
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>
>// clipped_average(): calculates percentiles limits, and then makes another 
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>//                    pass over the data to calculate average with values inside the percentiles
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>let clipped_average = (T:(x: long), lowPercentile:double, upPercentile:double)
#^^^ source.kusto keyword.control.kusto
#   ^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto
#                             ^^^^ source.kusto storage.type.kusto
#                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto
>{
#^^ source.kusto
>   let high = toscalar(T | summarize percentiles(x, upPercentile));
#^^^ source.kusto
#   ^^^ source.kusto keyword.control.kusto
#      ^^^^^^^^ source.kusto
#              ^^^^^^^^ source.kusto support.function.kusto
#                      ^^^^^ source.kusto
#                           ^^^^^^^^^ source.kusto keyword.other.query.kusto
#                                    ^ source.kusto
#                                     ^^^^^^^^^^^ source.kusto support.function.kusto
#                                                ^^^^^^^^^^^^^^^^^^^^ source.kusto
>   let low = toscalar(T | summarize percentiles(x, lowPercentile));
#^^^ source.kusto
#   ^^^ source.kusto keyword.control.kusto
#      ^^^^^^^ source.kusto
#             ^^^^^^^^ source.kusto support.function.kusto
#                     ^^^^^ source.kusto
#                          ^^^^^^^^^ source.kusto keyword.other.query.kusto
#                                   ^ source.kusto
#                                    ^^^^^^^^^^^ source.kusto support.function.kusto
#                                               ^^^^^^^^^^^^^^^^^^^^^ source.kusto
>   T 
#^^^^^^ source.kusto
>   | where x > low and x < high
#^^^^^ source.kusto
#     ^^^^^ source.kusto keyword.other.query.kusto
#          ^^^^^^^^^ source.kusto
#                   ^^^ source.kusto keyword.other.operator.kusto
#                      ^^^^^^^^^^ source.kusto
>   | summarize avg(x) 
#^^^^^ source.kusto
#     ^^^^^^^^^ source.kusto keyword.other.query.kusto
#              ^ source.kusto
#               ^^^ source.kusto support.function.kusto
#                  ^^^^^ source.kusto
>};
#^^^ source.kusto
>range x from 1 to 100 step 1
#^^^^^ source.kusto meta.query.range.kusto keyword.other.query.kusto
#     ^ source.kusto meta.query.range.kusto
#      ^ source.kusto meta.query.range.kusto variable.other.column.kusto
#       ^ source.kusto meta.query.range.kusto
#        ^^^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#            ^ source.kusto meta.query.range.kusto
#             ^ source.kusto meta.query.range.kusto constant.numeric.kusto
#              ^ source.kusto meta.query.range.kusto
#               ^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                 ^ source.kusto meta.query.range.kusto
#                  ^^^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                     ^ source.kusto meta.query.range.kusto
#                      ^^^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                          ^ source.kusto meta.query.range.kusto
#                           ^ source.kusto meta.query.range.kusto constant.numeric.kusto
>| invoke clipped_average(5, 99)
#^^ source.kusto
#  ^^^^^^ source.kusto meta.query.invoke.kusto keyword.other.query.kusto
#        ^ source.kusto meta.query.invoke.kusto
#         ^^^^^^^^^^^^^^^ source.kusto meta.query.invoke.kusto entity.name.function.kusto
#                        ^ source.kusto
#                         ^ source.kusto constant.numeric.kusto
#                          ^^ source.kusto
#                            ^^ source.kusto constant.numeric.kusto
#                              ^^ source.kusto
>